### PR TITLE
Preserve existing XDG_CURRENT_DESKTOP

### PR DESCRIPTION
**Pull request by Vladimir-csp**
#### Describe your PR, what does it fix/add?

Preserves existing `XDG_CURRENT_DESKTOP`, only set if empty.  For compatibility with DMs/SMs.

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

My 1st cpp commit. Yay!

#### Is it ready for merging, or does it need work?

I have no idea

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -36,7 +36,10 @@ int main(int argc, char** argv) {
     setenv("XDG_BACKEND", "wayland", 1);
     setenv("_JAVA_AWT_WM_NONREPARENTING", "1", 1);
     setenv("MOZ_ENABLE_WAYLAND", "1", 1);
-    setenv("XDG_CURRENT_DESKTOP", "Hyprland", 1);
+
+    // Set XDG_CURRENT_DESKTOP if it was not set for us
+    if (!getenv("XDG_CURRENT_DESKTOP"))
+        setenv("XDG_CURRENT_DESKTOP", "Hyprland", 1);
```
Comment (vaxerski) [inline src/main.cpp:42]: you should also add a flag to not `unsetenv` it on exit if it wasn't overwritten

Comment (Vladimir-csp) [reply src/main.cpp:42]: Does this look right?
```
    // Set XDG_CURRENT_DESKTOP if it was not set for us
    bool xdgcdExisted = False;
    if (!getenv("XDG_CURRENT_DESKTOP")) {
        setenv("XDG_CURRENT_DESKTOP", "Hyprland", 1);
    } else {
        xdgcdExisted = True;
    }
```
I'm afraid, connecting this to `CCompositor::cleanEnvironment()` might be beyond my current knowledge of C++ which is practically nonexistent.

Comment (leiserfg) [reply src/main.cpp:42]: Maybe instead of storing `xdgcdExisted` it will be simpler to unset  XDG_CURRENT_DESKTOP only if it is `Hyprland`?

Comment (Vladimir-csp) [reply src/main.cpp:42]: If it came from desktop entry, this would be ambiguous.

Comment (vaxerski) [reply src/main.cpp:42]: I'll do it myself then

Comment (Vladimir-csp) [reply src/main.cpp:42]: Thanks!

**Review by vaxerski — APPROVED** [review]

**Pull request merged**